tests/test_interpolation.py

```diff
@@ -1099,9 +1099,186 @@ def test_inject_subdomain_sinc(self):
+        each interpolation uses a unique temporary variable.
+
+        Bug: Before fix, all interpolations used Symbol(name='sum'), causing
+        value contamination when ConditionalDimensions skipped some loops.
+
+        Fix: Each interpolation now uses Symbol(name=f'sum_{sf.name}').
+
+        This test creates 3 SparseFunctions interpolating from 3 different
+        SubDomains in a single Operator, ensuring no cross-contamination.
+        """
+        grid = Grid(shape=(12, 12), extent=(10., 10.))
+
+        # Create 3 non-overlapping subdomains
+        class LeftDomain(SubDomain):
+            name = 'left'
+
+            def define(self, dimensions):
+                x, y = dimensions
+                return {x: ('left', 3), y: y}
+
+        class MiddleDomain(SubDomain):
+            name = 'middle'
+
+            def define(self, dimensions):
+                x, y = dimensions
+                return {x: ('middle', 3, 3), y: y}
+
+        class RightDomain(SubDomain):
+            name = 'right'
+
+            def define(self, dimensions):
+                x, y = dimensions
+                return {x: ('right', 3), y: y}
+
+        left = LeftDomain(grid=grid)
+        middle = MiddleDomain(grid=grid)
+        right = RightDomain(grid=grid)
+
+        # Register subdomains with grid
+        grid._subdomains = grid._subdomains + (left, middle, right)
+
+        # Functions on different subdomains with distinct values
+        f_left = Function(name='f_left', grid=left)
+        f_middle = Function(name='f_middle', grid=middle)
+        f_right = Function(name='f_right', grid=right)
+
+        # Initialize with different constant values
+        f_left.data[:] = 10.0
+        f_middle.data[:] = 20.0
+        f_right.data[:] = 30.0
+
+        # 3 SparseFunctions with points in different regions
+        sr_left = SparseFunction(name='sr_left', grid=grid, npoint=2)
+        sr_middle = SparseFunction(name='sr_middle', grid=grid, npoint=2)
+        sr_right = SparseFunction(name='sr_right', grid=grid, npoint=2)
+
+        # Points: one inside each subdomain, one outside
+        # Inside left, outside
+        sr_left.coordinates.data[:] = np.array([[1.5, 5.0], [8.5, 5.0]])
+        # Inside middle, outside
+        sr_middle.coordinates.data[:] = np.array([[5.5, 5.0], [1.5, 5.0]])
+        # Inside right, outside
+        sr_right.coordinates.data[:] = np.array([[9.5, 5.0], [5.5, 5.0]])
+
+        # Create operator with all 3 interpolations
+        rec_left = sr_left.interpolate(f_left)
+        rec_middle = sr_middle.interpolate(f_middle)
+        rec_right = sr_right.interpolate(f_right)
+
+        op = Operator([rec_left, rec_middle, rec_right])
+        op.apply()
+
+        # Verify: Each sparse function should have the correct value for points
+        # inside its subdomain, and 0.0 (NOT garbage!) for points outside
+        rank = grid.distributor.myrank
+
+        # Check all values are finite (catches garbage like 1e30)
```

Comment:
We already check `isclose` in existing tests. Really struggling to see what this test adds and it doesn't check what it claims to in the name/docstring. It is also independent of any SubDomains, so unsure why they are added.